cluster/gce/gci/configure-helper.sh

```diff
@@ -2076,6 +2076,42 @@ function start-fluentd-resource-update {
   wait-for-apiserver-and-update-fluentd &
 }
 
+# VolumeSnapshot CRDs are installed by cluster addon manager, which may
+# not be available at this point. Run this as a background process.
+function wait-for-apiserver-and-update-volumesnapshot {
+  # Wait until volumesnapshot CRDs is in place.
+  until kubectl get volumesnapshotclasses.snapshot.storage.k8s.io
+  do
+    sleep 10
+  done
+
+  until kubectl get volumesnapshotcontents.snapshot.storage.k8s.io
+  do
+    sleep 10
+  done
+
+  until kubectl get volumesnapshots.snapshot.storage.k8s.io
+  do
+    sleep 10
+  done
+
+  until kubectl get clusterrolebinding common-snapshot-controller-role
+  do
+    sleep 10
+  done
+
+  until kubectl get statefulset common-snapshot-controller | grep common-snapshot-controller | grep "1/1"
```

Comment:
I didn't realize this function is called in the background. In that case what does waiting for all these things help with?

Reply:
I followed how fluentd resource is updated:

https://github.com/kubernetes/kubernetes/blob/master/cluster/gce/gci/configure-helper.sh#L2075